app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,64 @@
+class RecipesController < ApplicationController
+  before_action :set_recipe, only: %i[show edit update destroy]
+
+  # GET /recipes
+  def index
+    @recipes = Recipe.all
+  end
+
+  # GET /recipes/1
+  def show; end
+
+  # GET /recipes/new
+  def new
+    @recipe = Recipe.new
+  end
+
+  # GET /recipes/1/edit
+  def edit; end
+
+  # POST /recipes or /recipes.json
+  def create
+    @recipe = Recipe.new(recipe_params)
+
```

Comment:
I think it is better if the recipe form does not request to add user_id before it can be created, users of the page might not know their user_id, so I think it's best to include this line `@recipe.user_id = current_user.id` in the create method instead.